src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -58,13 +60,26 @@ class SessionService(
     val today = LocalDate.now()
     val requestedBookableStartDate = today.plusDays(noticeDaysMin ?: policyNoticeDaysMin)
     val requestedBookableEndDate = today.plusDays(noticeDaysMax ?: policyNoticeDaysMax)
+    var prisonerWing: String?
 
-    val sessionTemplates = sessionTemplateRepository.findValidSessionTemplatesByPrisonId(
+    // get all the sessions for that prison - wing and non wing based
+    var sessionTemplates = sessionTemplateRepository.findValidSessionTemplatesByPrisonId(
       prisonId,
       requestedBookableStartDate,
       requestedBookableEndDate
     )
 
+    // filter the sessionTemplates to use the ones that are eligible for all wings and sessionTemplates for that wing
+    prisonerId?.also {
+      prisonerWing = getPrisonerWing(it)
+
+      // return all sessions where prison wings list is empty OR
+      // sessions where wings contain the prisoners wing
+      sessionTemplates = sessionTemplates.filter { sessionTemplate ->
+        isSessionEligibleForAllWings(sessionTemplate) || (prisonerWing != null && isSessionEligibleForWing(sessionTemplate, prisonerWing))
```

Comment:
not sure about this,... is it an OR? I think I'm trying to work this out while I type.. if a wing has a certain session  template for it should it ignore other session templates for the same prison that does not have a wing configuration?